evalml/pipelines/utils.py

```diff
@@ -397,8 +397,8 @@ def make_pipeline(
              An empty dictionary or None implies using all default values for component parameters.
          sampler_name (str): The name of the sampler component to add to the pipeline. Only used in classification problems.
              Defaults to None
-         extra_components (list[ComponentBase]): List of extra components to be added after preprocessing components. Defaults to None.
-         extra_components_position (str): Where to put extra components. Defaults to "before_preprocessing" and any other value will put components after preprocessing components.
+         extra_components_before (list[ComponentBase]): List of extra components to be added before preprocessing components. Defaults to None.
```

Comment:
Revised this API here to just have two lists for components in both locations. I was thinking about changing it to a component to index mapping but thought it would be too complicated (I would need to figure out index locations and how they would change as we insert each component).

Reply:
I think this is a solid change. I think you should let the use cases drive the API.  Do we currently have (or foresee) a reason to be able to inject components at different locations?

Reply:
I don't see any short term usage other than in `DefaultAlgorithm` but I can imagine sometime in the future we would want more control over where exact components go in our pipelines. I agree that we should let the use cases drive the API so let's leave this for now!